StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c

```diff
@@ -117,10 +117,9 @@ MmIplBuildFvHob (
   EFI_HOB_FIRMWARE_VOLUME  *FvHob;
   UINT16                   HobLength;
 
-  ASSERT (Hob != NULL);

```

Comment:
please clearly comment what's the problem happen in the commit message. And also consider how to avoid the Undefined Behavior (UB):

MmIplBuildMmCoreModuleHob (
FoundationHobList + UsedSize, /// FoundationHobList can be NULL, NULL+value is the Undefined Behavior
&HobLength,
MmCoreFileName,
MmCoreImageAddress,
MmCoreImageSize,
MmCoreEntryPoint
);

Could we change as below:
FoundationHobList != NULL ? FoundationHobList + UsedSize : NULL

Reply:
Add the description to commit message now.
It should be enough that check the pointer before using it if it is NULL.

Reply:
hongbin, i think "FoundationHobList != NULL ? FoundationHobList + UsedSize : NULL" is better added for the clearly avoid the undefined behavior.